tests/ctst/steps/bucket-policies/utils.ts

```diff
@@ -0,0 +1,500 @@
+type ActionPermissionsType = {
+    action: string;
+    permissions: string[];
+    subAuthorizationChecks?: boolean;
+    expectedResultOnAllowTest?: string;
+    excludePermissionOnBucketObjects?: boolean,
+    excludePermissionsOnBucket?: boolean,
+    useWildCardBucketName?: boolean,
+    needsSetup?: boolean,
+};
+
+const needObjectLock = [
+    'GetObjectLegalHold',
+    'PutObjectLegalHold',
+    'GetObjectRetention',
+    'PutObjectRetention',
+    'PutObjectVersionLegalHold',
+    'PutObjectVersionRetention',
+    'GetObjectVersionLegalHold',
+    'GetObjectVersionRetention',
+];
+
+const needObject = [
+    'PutObjectLegalHold',
+    'PutObjectRetention',
+    'PutObjectTagging',
+    'PutObjectAcl',
+    'PutObject',
+    'GetObject',
+    'GetObjectAcl',
+    'GetObjectLegalHold',
+    'GetObjectRetention',
+    'GetObjectTagging',
+    'DeleteObject',
+    'DeleteObjectTagging',
+    'DeleteObjects',
+    'HeadObject',
+    'UploadPart',
+    'UploadPartCopy',
+    'DeleteObjectVersion',
+    'DeleteObjectVersionTagging',
+    'GetObjectVersion',
+    'GetObjectVersionAcl',
+    'GetObjectVersionTagging',
+    'PutObjectVersionAcl',
+    'PutObjectVersionTagging',
+    'PutObjectVersionRetention',
+    'PutObjectVersionLegalHold',
+];
+
+const needVersioning = [
+    'PutBucketReplication',
+    'ListObjectVersions',
+    'DeleteObjectVersion',
+    'DeleteObjectVersionTagging',
+    'GetObjectVersion',
+    'GetObjectVersionAcl',
+    'GetObjectVersionTagging',
+    'PutObjectVersionAcl',
+    'PutObjectVersionTagging',
+    'PutObjectVersionRetention',
+    'PutObjectVersionLegalHold',
+];
+
+const actionPermissions: ActionPermissionsType[] = [
+    // Create bucket is not checking any bucket policy
+    // {
+    //     action: 'CreateBucket',
+    //     permissions: ['s3:CreateBucket'],
+    //     expectedResultOnAllowTest: 'BucketAlreadyOwnedByYou',
+    //     excludePermissionOnBucketObjects: true,
+    //     useWildCardBucketName: true,
+    // },
+    {
+        action: 'PutBucketAcl',
+        permissions: ['s3:PutBucketAcl'],
+        expectedResultOnAllowTest: 'MalformedXML',
+        excludePermissionOnBucketObjects: true,
+    },
+    {
+        action: 'PutBucketEncryption',
+        permissions: ['s3:PutEncryptionConfiguration'],
+        expectedResultOnAllowTest: 'MalformedXML',
```

Comment:
same https://github.com/scality/Integration/blob/development/7.10/tests/bucket-policy/utils/s3APIConfigurations.js#L140

Reply:
Same for this API, this error is returned after authz check, no need to test it fully: https://github.com/scality/cloudserver/blob/2596f3fda8265b981e0241dcf3a66a87379efb79/lib/api/bucketPutEncryption.js#L31